app/actions/index.js

```diff
@@ -129,49 +130,65 @@ const checkOsmMeta = (url, getState, cb) => {
 };
 
 export const syncSurveyData = survey => (dispatch, getState) => {
+  console.log("syncSurveyData");
   const { id, target } = survey;
   const url = `http://${target.address}:${target.port}`;
 
-  checkOsmMeta(url, getState, (err, shouldSync, areaOfInterest) => {
+  checkOsmMeta(url, getState, (err, shouldImportOsm, areaOfInterest) => {
     if (err) return console.warn(err);
-    if (!shouldSync) {
-      return dispatch({
-        type: types.FINISHED_SYNCING_SURVEY_DATA,
-        id
-      });
-    }
+    console.log("shouldImportOsm", shouldImportOsm);
 
     dispatch({
       type: types.SYNCING_SURVEY_DATA,
       id: id
     });
 
+    if (shouldImportOsm) {
+      osm.replicate(
+        target,
+        {
+          progressFn
+        },
+        err => {
+          if (err) {
+            return dispatch({
+              type: types.SYNCING_SURVEY_DATA_FAILED,
+              id
+            });
+          }
+
+          console.log("osm.replicate", err);
+          dispatch({
+            type: types.FINISHED_SYNCING_SURVEY_DATA,
+            id
+          });
+
+          dispatch({
+            type: types.SET_AREA_OF_INTEREST,
+            areaOfInterest
+          });
+        }
+      );
+    }
+
     const progressFn = i => {
+      console.log("progress", i);
       dispatch({
         type: types.SYNCING_SURVEY_DATA_PROGRESS,
         progress: i,
         id: id
       });
     };
+  });
+};
 
-    osm.replicate(target, { progressFn }, err => {
-      if (err) {
-        return dispatch({
-          type: types.SYNCING_SURVEY_DATA_FAILED,
-          id
-        });
-      }
-
-      dispatch({
-        type: types.FINISHED_SYNCING_SURVEY_DATA,
-        id
-      });
-
-      dispatch({
-        type: types.SET_AREA_OF_INTEREST,
-        areaOfInterest
-      });
-    });
+export const destroyAllData = () => dispatch => {
+  console.log("destroy all data");
+  dispatch({ type: types.CLEAR_LOCAL_SURVEYS });
+  dispatch({ type: types.CLEAR_REMOTE_SURVEYS });
```

Comment:
Another way to do [some] of this is to call `.purge()` on the result of `persistStore()` in https://github.com/osmlab/field-data-collection/blob/master/app/config/store.js#L20-L23 That's how I reset survey definition state.